contracts/gateway.fc

```diff
@@ -302,17 +302,20 @@ cell auth::ecdsa::external(slice message, slice expected_evm_address) inline {
 
     int tx_fee = get_gas_fee_workchain(gas::withdraw);
 
+    throw_if(error::insufficient_value, state::total_locked < (amount + tx_fee));
+
```

Comment:
nit: as this is a non obvious check we could add a comment describing the corner case.